idl/internal/lex.rl

```diff
@@ -301,10 +302,14 @@ func (lex *lexer) Lex(out *yySymType) int {
 
                 var str string
                 var err error
-                if len(bs) > 0 && bs[0] == '\'' {
-                    str, err = UnquoteSingleQuoted(bs)
-                } else {
-                    str, err = strconv.Unquote(string(bs))
+
+                str, err = Unescape(bs)
+                if err == nil && !strings.ContainsAny(str, "\r\n") {
```

Comment:
I added this new condition to prevent strings containing these control characters from being passed to `strconv.Unquote()`, which will always error in these cases.

This is a behavior change. If that's not desired, we have two options:

1. Roll our own `Unquote()` that can handle these cases.
2. Stop expanding `\r` and `\n` in `Unescape()`, meaning we'd only have partial support for escape sequences.

Reply:
For (2), would it be partial support given that strconv.Unquote will still expand `\r`, `\n`, `\t`, etc.?
Our issue seems to be that strconv.Unquote doesn't handle `\'`, so really, that's all we need to unescape.